backend/src/main/java/com/redhat/cloud/notifications/recipients/rbac/AuthRequestFilter.java

```diff
@@ -1,25 +1,71 @@
 package com.redhat.cloud.notifications.recipients.rbac;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.eclipse.microprofile.config.Config;
 import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.logmanager.Level;
 
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
 import javax.ws.rs.client.ClientRequestContext;
 import javax.ws.rs.client.ClientRequestFilter;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.util.Base64;
+import java.util.Map;
+import java.util.logging.Logger;
 
+@ApplicationScoped
 class AuthRequestFilter implements ClientRequestFilter {
 
-    String authInfo;
-    String secret;
-    String application;
+    static final String RBAC_SERVICE_TO_SERVICE_APPLICATION_KEY = "rbac.service-to-service.application";
+    static final String RBAC_SERVICE_TO_SERVICE_APPLICATION_DEFAULT = "notifications";
 
-    AuthRequestFilter() {
+    static final String RBAC_SERVICE_TO_SERVICE_SECRET_MAP_KEY = "rbac.service-to-service.secret-map";
+    static final String RBAC_SERVICE_TO_SERVICE_SECRET_MAP_DEFAULT = "{}";
+
+    // used by dev to by pass the service to service token: Uses user:password format
+    static final String RBAC_SERVICE_TO_SERVICE_DEV_EXCEPTIONAL_AUTH_KEY = "rbac.service-to-service.exceptional.auth.info";
+
+    private static class Secret {
+        public String secret;
+    }
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    private String authInfo;
+    private String secret;
+    private String application;
+
+    private final Logger log = Logger.getLogger(this.getClass().getName());
+
+    @PostConstruct
+    void init() {
         Config config = ConfigProvider.getConfig();
-        secret = config.getOptionalValue("rbac.service-to-service.secret", String.class).orElse("addme");
-        application = config.getOptionalValue("rbac.service-to-service.application", String.class).orElse("notifications");
-        String tmp = System.getProperty("rbac.service-to-service.exceptional.auth.info");
+
+        application = config.getOptionalValue(RBAC_SERVICE_TO_SERVICE_APPLICATION_KEY, String.class).orElse(RBAC_SERVICE_TO_SERVICE_APPLICATION_DEFAULT);
+        Map<String, Secret> rbacServiceToServiceSecretMap;
+
+        try {
+            rbacServiceToServiceSecretMap = objectMapper.readValue(
+                    config.getOptionalValue(RBAC_SERVICE_TO_SERVICE_SECRET_MAP_KEY, String.class).orElse(RBAC_SERVICE_TO_SERVICE_SECRET_MAP_DEFAULT),
+                    new TypeReference<>() { }
+            );
+        } catch (JsonProcessingException jsonProcessingException) {
+            log.log(Level.ERROR, "Unable to load Rbac service to service secret map, defaulting to empty map");
```

Comment:
Logging the exception may help understanding why it failed:
```suggestion
            log.log(Level.ERROR, "Unable to load Rbac service to service secret map, defaulting to empty map", jsonProcessingException);
```